### PR TITLE
Enable ffmpeg support

### LINKED ISSUE #27: ffmpeg support
Add support for video recording

image

### PR DESCRIPTION
**Pull request by binaryfields**
Close #27

### CODE REVIEW

**Comment by binaryfields** [comment]
Fixes #27

**Comment by Brog33** [comment]
could you make a new build so i can test it?

**Comment by hfiguiere** [comment]
bot, build

**Comment by hfiguiere** [comment]
bot, build

**Comment by hfiguiere** [comment]
I'll test and merge

**Comment by hfiguiere** [comment]
Close #27

**Comment by hfiguiere** [comment]
Actually this doesn't work and nobody reported it.

Start VICE and it terminates early.

If you start wth `--verbose` you get:

```
Cannot set string resource 'FFMPEGFormat' to default 'mp4'
Cannot set defaults.
```

ffmpeg needs to be built with mp4 support.

File: net.sf.VICE.yml [hunk 1]
```diff
@@ -10,54 +10,144 @@ finish-args:
   - --socket=pulseaudio
   - --socket=x11
   - --metadata=X-DConf=migrate-path=/net/sf/VICE/
+add-extensions: 
+  - org.freedesktop.Platform.ffmpeg-full: 
+    directory: lib/ffmpeg
+    version: 21.08
+    add-ld-path: .
```
Comment (hfiguiere) [inline net.sf.VICE.yml:17]:
```suggestion
    org.freedesktop.Platform.ffmpeg-full: 
        directory: lib/ffmpeg
        version: '21.08'
        add-ld-path: .
```

This is the proper syntax.

It even complain with `(flatpak-builder:2): Json-WARNING **: 02:44:55.926: Failed to deserialize "add-extensions" property of type "gpointer" for an object of type "BuilderManifest"`

Also `(flatpak-builder:2): flatpak-builder-WARNING **: 02:44:55.925: 16:14: '21.08' will be parsed as a number by many YAML parsers`

(YAML is amazing)

File: net.sf.VICE.yml [hunk 2]
```diff
@@ -10,54 +10,144 @@ finish-args:
...
+cleanup-commands:
+  - mkdir -p /app/lib/ffmpeg
```
Comment (hfiguiere) [inline net.sf.VICE.yml:19]:
```suggestion
  - --env=LD_LIBRARY_PATH=/app/lib/codecs/lib
```

Actually all you need is this.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
With the suggested fix it should work.

File: net.sf.VICE.yml (hunk 1 above)
Comment (hfiguiere) [reply net.sf.VICE.yml:17]: Just ignore this, see suggestio below.

**Pull request closed without merging**